### PR TITLE
Added device group data to /devicesview endpoint.

### PR DESCRIPTION
**Pull request by AaronH88**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

**Review by thom-at-redhat — APPROVED** [review]

**Review by thom-at-redhat — CHANGES_REQUESTED** [review]
Clearing approval due to timing

**Review by thom-at-redhat — APPROVED** [review]

File: pkg/models/devices.go [hunk 1]
```diff
@@ -37,17 +37,17 @@ type DeviceViewList struct {
...
-	ImageName       string `json:"ImageName"`
-	LastSeen        string `json:"LastSeen"`
-	UpdateAvailable bool   `json:"UpdateAvailable"`
-	Status          string `json:"Status"`
-	ImageSetID      uint   `json:"ImageSetID"`
+	DeviceID        uint     `json:"DeviceID"`
+	DeviceName      string   `json:"DeviceName"`
+	DeviceUUID      string   `json:"DeviceUUID"`
+	DeviceGroupID   []uint   `json:"DeviceGroupID"`
+	DeviceGroupName []string `json:"DeviceGroupName"`
```
Comment (ldjebran) [inline pkg/models/devices.go:44]: maybe create a struct 
```go
type DeviewDevioceGroup struct {
    ID uint
    Name string
}
```
and after 
```go
type DeviceView struct {
    ...
    DeviceGroups []DeviewDevioceGroup
    ...
```

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
43c9ec0f-03e7-473f-b152-765bd3885977
ccfbc295-4059-4d5b-b7a7-2f838f2963cf

File: pkg/models/devices.go (hunk 1 above)
Comment (thom-at-redhat) [reply pkg/models/devices.go:44]: fixing typos:
`type DeviceDeviceGroup struct {
    ID uint
    Name string
}`
and after
`type DeviceView struct {
    ...
    DeviceGroups []DeviceDeviceGroup
    ...`

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
35de4c0d-c575-4a2f-a1b1-cd472228cd33
ca5101df-b316-49fc-89a7-c77906514714

File: pkg/services/devices.go [hunk 2]
```diff
@@ -618,10 +618,24 @@ func (s *DeviceService) GetDevicesView(limit int, offset int, tx *gorm.DB) (*mod
 	}
 
 	var storedDevices []models.Device
-	if res := tx.Limit(limit).Offset(offset).Where("account = ?", account).Preload("UpdateTransaction").Find(&storedDevices); res.Error != nil {
+	if res := tx.Limit(limit).Offset(offset).Where("account = ? AND image_id IS NOT NULL", account).Preload("UpdateTransaction").Preload("DevicesGroups").Find(&storedDevices); res.Error != nil {
```
Comment (ldjebran) [inline pkg/services/devices.go:621]: we do not need "AND image_id IS NOT NULL" any more , as we have a more general get device view the main filter controlled by tx

Comment (ldjebran) [reply pkg/services/devices.go:621]: the "AND image_id IS NOT NULL" is in the router now

File: pkg/models/devices.go (hunk 1 above)
Comment (AaronH88) [reply pkg/models/devices.go:44]: I understand this point, However, we currently do not need to keep the relationship between the ID and the Name in this view. We just need to display the names in the table. 

Is keeping this relationship something you think we should have in this view?

Regardless, the return json object will not change, unless I further indent the return json object. 

e.g.

```
data": {
        "total": 28,
        "devices": [
            {
                "DeviceID": 1,
                "DeviceName": "",
                "DeviceUUID": "",
                "ImageID": 1,
                "ImageName": "",
                "LastSeen": "0001-01-01 00:00:00 +0000 UTC",
                "UpdateAvailable": false,
                "Status": "RUNNING",
                "ImageSetID": 0,
                "deviceGroupData": [
                  {
                    "DeviceGroupID": null,
                    "DeviceGroupName": null,
                }
                ]
            },
```

Is this what you want?

File: pkg/services/devices.go (hunk 2 above)
Comment (AaronH88) [reply pkg/services/devices.go:621]: removed :smile:

**Pull request merged**